x-pack/plugins/security_solution/server/lib/entity_analytics/risk_score/risk_score_data_client.ts

```diff
@@ -161,4 +165,43 @@ export class RiskScoreDataClient {
       throw error;
     }
   }
+  /**
+   * Ensures that configuration migrations for risk score indices are seamlessly handled across Kibana upgrades.
+   *
+   * Upgrades:
+   * - Migrating to 8.12+ requires a change to the risk score latest transform index's 'dynamic' setting to ensure that
+   * unmapped fields are allowed within stored documents.
+   *
+   */
+  public async upgrade() {
```

Comment:
NIT: I would name this function differently because it doesn't always upgrade. Something like `tryToUpgrade` or `upgradeIfNeeded`... naming is hard 😓

Reply:
Addressed, thank you both for the suggestion! 👍 https://github.com/elastic/kibana/pull/174474/commits/41d6e044a22d15c8daf146311e9730667832a86b